### PR TITLE
Convert asserts in unittest to pytest

### PR DESCRIPTION
**Pull request by pcmoritz**
This is the first step to converting our tests from unittest to pytest, rewriting all the assertions.

### CODE REVIEW

**Comment by shaneknapp** [comment]
test this please

File: test/actor_test.py [hunk 1]
```diff
@@ -34,46 +34,40 @@ def get_values(self, arg0, arg1=2, arg2="b"):
...
-        self.assertEqual(
-            ray.get(actor.get_values.remote(0, arg2="d")), (1, 4, "cd"))
-        self.assertEqual(
-            ray.get(actor.get_values.remote(0, arg2="d", arg1=0)),
-            (1, 2, "cd"))
-        self.assertEqual(
-            ray.get(actor.get_values.remote(arg2="d", arg1=0, arg0=2)),
-            (3, 2, "cd"))
+        assert ray.get(actor.get_values.remote(0, arg2="d")) == (1, 4, "cd")
+        assert ray.get(actor.get_values.remote(0, arg2="d", arg1=0)) == \
```
Comment (robertnishihara) [inline test/actor_test.py]: Shouldn't use `\`. Instead use multi-line continuations with parentheses.

File: test/actor_test.py [hunk 2]
```diff
@@ -600,10 +588,9 @@ def get_value(self):
             actor = Actor1.remote(x)
             return ray.get([actor.get_value.remote() for _ in range(n)])
 
-        self.assertEqual(ray.get(f.remote(3, 1)), [3])
-        self.assertEqual(
-            ray.get([f.remote(i, 20) for i in range(10)]),
-            [20 * [i] for i in range(10)])
+        assert ray.get(f.remote(3, 1)) == [3]
+        assert ray.get([f.remote(i, 20) for i in range(10)]) == \
```
Comment (robertnishihara) [inline test/actor_test.py]: `\`

File: test/actor_test.py [hunk 3]
```diff
@@ -1085,9 +1069,8 @@ def locations_to_intervals_for_many_tasks():
         # Run a bunch of GPU tasks.
         locations_to_intervals = locations_to_intervals_for_many_tasks()
         # Make sure that all GPUs were used.
-        self.assertEqual(
-            len(locations_to_intervals),
-            num_local_schedulers * num_gpus_per_scheduler)
+        assert len(locations_to_intervals) == \
```
Comment (robertnishihara) [inline test/actor_test.py]: `\`

File: test/actor_test.py [hunk 4]
```diff
@@ -1122,17 +1104,16 @@ def locations_to_intervals_for_many_tasks():
         # Run a bunch of GPU tasks.
         locations_to_intervals = locations_to_intervals_for_many_tasks()
         # Make sure that all but 11 of the GPUs were used.
-        self.assertEqual(
-            len(locations_to_intervals),
-            num_local_schedulers * num_gpus_per_scheduler - 1 - 3)
+        assert len(locations_to_intervals) == \
```
Comment (robertnishihara) [inline test/actor_test.py]: `\`

Comment (robertnishihara) [reply test/actor_test.py]: many other places as well

**Comment by robertnishihara** [comment]
Thanks for doing this!

**Review by robertnishihara — APPROVED** [review]

**Pull request merged**